enterprise/server/remote_execution/containers/firecracker/firecracker.go

```diff
@@ -1860,10 +1860,15 @@ func (c *FirecrackerContainer) SendExecRequestToGuest(ctx context.Context, cmd *
 	health := hlpb.NewHealthClient(conn)
 
 	defer container.Metrics.Unregister(c)
+	var lastObservedStatsMutex sync.Mutex
 	var lastObservedStats *repb.UsageStats
 	statsListener := func(stats *repb.UsageStats) {
 		container.Metrics.Observe(c, stats)
-		lastObservedStats = stats
+		{
+			lastObservedStatsMutex.Lock()
+			defer lastObservedStatsMutex.Unlock()
+			lastObservedStats = stats
+		}
```

Comment:
nit: since there's no error checking that might disrupt the control flow and require the defer, this can be simplified to: 

statsListener := func(stats *repb.UsageStats) {
		container.Metrics.Observe(c, stats)
		lastObservedStatsMutex.Lock()
		lastObservedStats = stats
		lastObservedStatsMutex.Unlock()
}


(and same below)

Reply:
kk, wasn't sure of the preferred style. Done.